### PR TITLE
Don't truncate volume for stacks

### PR DESCRIPTION
**Pull request by mugling**
See discussion in #15235

### CODE REVIEW

File: src/item.cpp [hunk 1]
```diff
@@ -2385,8 +2385,7 @@ int item::volume( bool integral ) const
 
     // For items counted per charge the above volume is per stack so adjust dependent upon charges
     if( count_by_charges() || made_of( LIQUID ) ) {
-        ret *= charges;
-        ret /= type->stack_size;
+        ret = ceil( std::min( double( charges ) / type->stack_size, 1.0 ) * ret );
```
Comment (kevingranade) [inline src/item.cpp]: If charges / stack_size is greater than 1, why do we want it reduced to 1.0?

**Comment by mugling** [comment]
I think I was trying (incorrectly) to avoid division by zero. I'm just building that branch now but possibly something along the lines of:
```
ret = ceil( charges / std::min( double( type->stack_size ), 1.0 ) * ret );
```

**Comment by mugling** [comment]
Nope, that's subtly broken a well

**Comment by mugling** [comment]
2c8f73a adds stack size to `item::info` as this becomes more relevant after this PR. It also removes the misleading text 'Default stack size' from ammo to avoid confusion

File: src/item.cpp [hunk 2]
```diff
@@ -2385,8 +2387,8 @@ int item::volume( bool integral ) const
 
     // For items counted per charge the above volume is per stack so adjust dependent upon charges
     if( count_by_charges() || made_of( LIQUID ) ) {
-        ret *= charges;
-        ret /= type->stack_size;
+        auto sk = std::max( type->stack_size, 1 );
```
Comment (kevingranade) [inline src/item.cpp]: Big nope, auto is great for iterators, but please don't use it for basic types.

**Pull request merged**